phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -2037,7 +2049,10 @@ public void createTable(RpcController controller, CreateTableRequest request,
                     // view's property in case they are different from the parent
                     ViewUtil.addTagsToPutsForViewAlteredProperties(tableMetadata, parentTable);
                 }
-
+                //set the last DDL timestamp to the current server time since we're creating the
+                // table
+                tableMetadata.add(MetaDataUtil.getLastDDLTimestampUpdate(tableKey,
+                    clientTimeStamp, EnvironmentEdgeManager.currentTimeMillis()));
```

Comment:
Do we want to restrict this to just tables and views i.e. 'u' and 'v' table_types? The upgrade code only adds a ts for existing tables and views, but not indexes and SYSTEM tables, but here we do it for all types. There will be inconsistency in that case between an index created before upgrading (no ts) vs an index created after upgrading (has ts), not to mention fresh clusters will have a ts for all entities.

Reply:
Thanks, didn't realize those other types also went through that code path. Will add the restriction.